Source/WebCore/html/CanvasBase.cpp

```diff
@@ -109,7 +111,7 @@ void CanvasBase::makeRenderingResultsAvailable()
 {
     if (auto* context = renderingContext()) {
         context->paintRenderingResultsToCanvas();
-        if (auto noiseInjectionHashSalt = shouldInjectNoiseBeforeReadback() ? scriptExecutionContext()->noiseInjectionHashSalt() : std::nullopt)
+        if (auto noiseInjectionHashSalt = m_canvasNoiseHashSalt)
             m_canvasNoiseInjection.postProcessDirtyCanvasBuffer(buffer(), *noiseInjectionHashSalt);
```

Comment:
(Ditto, re: just using `m_canvasNoiseHashSalt`)

```suggestion
        if (m_canvasNoiseHashSalt)
            m_canvasNoiseInjection.postProcessDirtyCanvasBuffer(buffer(), * m_canvasNoiseHashSalt);
```